### ISSUE TITLE
accessing chart.series may break chart with series whose order is smaller than idx

### ISSUE THREAD

**Issue by huandzh** [scanny/python-pptx#123]
Hi,

When modifying a chart with a series whose order is smaller than idx, accessing series property of the chart will change `idx` and `order` of the series. This may break the chart. (See example)

After some dig ups, I have found that the getter of property **CT_ChartSpace.sers** changes idx and order (See https://github.com/scanny/python-pptx/blob/master/pptx/oxml/chart/chart.py#L104-L119). 

I guess we need a read-only getter of `sers` in oxml module and add a method to do the sorting and correcting.

Any ideas?
# Example:

``` python
>>> import pptx
>>> prs = pptx.Presentation('sample.pptx') # a working chart can be edited with MS Office
>>> chart = prs.slides[0].shapes[0].chart
>>> for ser in chart._chartSpace.xpath('//c:ser'):
...    print ser.idx.val, ser.order.val
5 0
0 1
1 2
>>>  first_series = chart.series[0] #just access this series
>>> prs.save('sample_broken.pptx') # get a broken file 
```

`idx` and `order` **changed unexpectedly**

``` python
>>> for ser in chart._chartSpace.xpath('//c:ser'):
...    print ser.idx.val, ser.order.val
2 2 <----
0 1
1 2 <----
```

**Comment by scanny** [comment]
Okay, this should be sorted out in the upcoming release.

This problem was related to the `c:ser/c:idx` and `c:ser/c:order` element semantics and that was just plain not documented anywhere. In the end, sorting it out required experimentation. Once I discovered the pattern it seemed almost self-evident. Funny how that works :)

The short version is this:

* both `c:idx` and `c:order` are required children of `c:ser`.

* The value of `c:idx` acts like an identifier for the series. Its sequence is not significant and gaps in the numbering turn out to be commonplace. The safest bet is to never change them and always create a new one as one more than the max one that's there, across the whole chart, even if it has multiple plots (i.e. multiple xChart elements, e.g. barChart, lineChart, etc.).

  There are situations where a series needs to be referenced from other parts of the XML, like to map it to a legend entry maybe. This is done using the idx value. So now we just keep them unique and let them be.

* The value of `c:order` determines the ordering of the series *within an xChart element*. The ordering of plots (xChart elements) when a chart has more than one is determined internally by PowerPoint. `c:order` values can bounce around between xChart elements, like 0, 2, 4 in the first one and 1, 3 in the second one. Doesn't matter. They have to be unique and series are sorted in that order within the xChart, but otherwise the overall sequence in the chart is not significant.

Anyway, all that is fixed up now and should drop in a week or two in release 0.6.2.

**Comment by huandzh** [comment]
Thanks! Good to hear that 👍

**Comment by scanny** [comment]
This should be fixed in release 0.6.2, out later today.
